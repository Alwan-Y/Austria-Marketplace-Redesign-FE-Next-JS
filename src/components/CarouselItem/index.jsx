import classNames from 'classnames';

const CarouselItem = ({
  active,
  title,
  productDescription,
  imgPath,
  classImage,
}) => {
  return (
    <div className={classNames('carousel-item', { ['active']: active })}>
      <div className="row pt-5 justify-content-center">
        <div className="col-9 col-sm-4 col-md-6 col-lg-5 ">
          <h1 className="mb-4 carousel__h1 mt-5">{title}</h1>
          <p className="mb-4 carousel__p">{productDescription}</p>
        </div>
        <div className="col-3 col-sm-6 col-md-4 col-lg-4 d-none d-sm-block offset-1">
          <img src={imgPath} className={classImage} />
        </div>
      </div>
    </div>
  );
};

export default CarouselItem;
