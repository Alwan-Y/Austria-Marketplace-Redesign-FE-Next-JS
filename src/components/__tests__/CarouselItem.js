import CarouselItem from '../commons/CarouselItem';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';

test('Carousel Item renders correctly', () => {
  const { container } = render(<CarouselItem></CarouselItem>);

  expect(container).toMatchSnapshot();
});

test('Carousel have class carousel-item', () => {
  const { container } = render(
    <CarouselItem className="carousel-item"></CarouselItem>
  );

  expect(container.firstChild).toHaveClass('carousel-item');
});

test('Carousel have class active carousel-item', () => {
  const { container } = render(
    <CarouselItem className="carousel-item " active="active"></CarouselItem>
  );

  expect(container.firstChild).toHaveClass('carousel-item active');
});

test('Carousel have class justify-content-center', () => {
  const { container } = render(
    <CarouselItem
      className="justify-content-center"
      active="active"
    ></CarouselItem>
  );

  expect(container.firstElementChild).toHaveClass('carousel-item active');
});
