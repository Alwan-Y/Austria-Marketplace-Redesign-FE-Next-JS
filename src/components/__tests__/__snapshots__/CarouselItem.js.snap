// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Item renders correctly 1`] = `
<div>
  <div
    class="carousel-item"
  >
    <div
      class="row pt-5 justify-content-center"
    >
      <div
        class="col-9 col-sm-4 col-md-6 col-lg-5"
      >
        <h1
          class="mb-4"
        />
        <p
          class="mb-4"
        />
        <a
          class="btn btn-warning text-white"
          href=""
        >
          Get It Now
        </a>
      </div>
      <div
        class="col-3 col-sm-6 col-md-4 col-lg-4 d-none d-sm-block offset-1"
      >
        <img />
      </div>
    </div>
  </div>
</div>
`;
